File: petclinic-frontend/src/features/visits/Emergency/Model/EmergencyRequestDTO.ts
import { UrgencyLevel } from './UrgencyLevel';

export interface EmergencyRequestDTO {
  visitDate: string;

  description: string;

  petId: string;

  practitionerId: string;

  petName: string;

  urgencyLevel: UrgencyLevel;

  emergencyType: string;
}

File: visits-service-new/src/main/java/com/petclinic/visits/visitsservicenew/PresentationLayer/Emergency/EmergencyRequestDTO.java
package com.petclinic.visits.visitsservicenew.PresentationLayer.Emergency;

import com.fasterxml.jackson.annotation.JsonFormat;
import com.petclinic.visits.visitsservicenew.DataLayer.Emergency.UrgencyLevel;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class EmergencyRequestDTO {
    @JsonFormat(pattern = "yyyy-MM-dd HH:mm")
    LocalDateTime visitDate;
    private String description;

    private String petId;
    private String practitionerId;

    private UrgencyLevel urgencyLevel;


    private String emergencyType;
}
